### DISCUSSION TITLE
Format date object in multiple documents - RangeError for null / empty object

### DISCUSSION THREAD

**Discussion by gyinshen** [js-fns/date-fns#2567] [Q&A]
Hi community, thanks for your great contribution. I'm using date-fns to convert ISO dates from my mongoDB database. The problem is that not every document have the date object which causes RangeError: Invalid time value error on my react app.

I'm displaying the documents using react table and tried to exclude documents which do not have a date object.

`  const columns = useMemo(
    () => [
      {
        Header: "All companies",
        columns: [
          {
            Header: "Founding Date",
            accessor: "companyFoundationDate",
            Cell: ({ value }) => { 
              if(!isValid(value) && value !== 'NaN') {
                return format(new Date(value), 'MM/yyyy')
                }
              }
          },
        ]
      },
    ],
    []
  );`

However, this does not work and continues to throw the same error without the if statement. I'm thinking that it's because the if statement doesn't work here. Do you think you can help me improve this?

Thanks!

**Comment by maximtop** [comment]
Shouldn't condition be without negation before isValid, like `(isValid(value) && value !== 'NaN')`?

**Reply by gyinshen** [reply]
> Shouldn't condition be without negation before isValid, like `(isValid(value) && value !== 'NaN')`?

Thanks for your reply. Yes it was an error. But changing it still yields the same RangeError error.
<img width="1120" alt="Screenshot 2021-07-29 at 2 50 55 PM" src="https://user-images.githubusercontent.com/25930732/127445407-31cc8f42-2742-4794-9cb0-0aaf542528f1.png">

**Reply by maximtop** [reply]
seems like on screenshot the error is caused by another part of your code, and there is no check for isValid

**Reply by gyinshen** [reply]
Sorry that I was not organized. The screenshot is wrong because I was experimenting with isDate too.

With the if statement, it will show Error: Cell(...): Nothing was returned from render. This usually means a return statement is missing. Or, to render nothing, return null.

`  const columns = useMemo(
    () => [
      {
        Header: "All royalties",
        columns: [
          {
            Header: "Start Date",
            accessor: "royaltyStartDate",
            Cell: ({ value }) => { 
                if(isValid(value) && value !== 'NaN') {
                  return format(new Date(value), 'MM/yyyy')
                }
                }
            },
        ]
      },
    ],
    []
  );`

<img width="1040" alt="Screenshot 2021-07-29 at 6 05 17 PM" src="https://user-images.githubusercontent.com/25930732/127473453-aeb45470-7372-4c78-9bc0-3ef131c0025b.png">

Without the if statement, it will show RangeError: Invalid time value error

`
  const columns = useMemo(
    () => [
      {
        Header: "All royalties",
        columns: [
          {
            Header: "Start Date",
            accessor: "royaltyStartDate",
            Cell: ({ value }) => { 
                  return format(new Date(value), 'MM/yyyy')
                }
            },
        ]
      },
    ],
    []
  );`

<img width="1024" alt="Screenshot 2021-07-29 at 6 09 27 PM" src="https://user-images.githubusercontent.com/25930732/127473887-a639ae3e-bbf3-4b92-9b70-463ab86721df.png">

I have another set of documents with similar structure and it's working perfectly with the code above (the 2nd one without if statement). The difference is that this set of documents all have date objects.

**Reply by maximtop** [reply]
The error states, that you should return null
![image](https://user-images.githubusercontent.com/17272769/127475927-795d6825-4c3d-4873-8d6e-01a137974775.png)
have you tried to return null in the else branch?

```
const columns = useMemo(() => [{
    Header: "All royalties",
    columns: [{
        Header: "Start Date",
        accessor: "royaltyStartDate",
        Cell: ({
            value
        }) => {
            if (isValid(value) && value !== 'NaN') {
                return format(new Date(value), 'MM/yyyy')
            } else {
                return null
            }
        }
    }, ]
}, ], []);
```

**Reply by gyinshen** [reply]
Thanks buddy. I tried return null. The error went away but react shows a blank column.

<img width="1109" alt="Screenshot 2021-07-29 at 6 46 33 PM" src="https://user-images.githubusercontent.com/25930732/127478978-85967470-1af6-496f-a717-0826a1e0f932.png">

**Reply by maximtop** [reply]
isValid checks if date is valid, and not a string
I think that you should parse your string before checking its validity
